libmscore/note.cpp

```diff
@@ -2004,12 +2005,9 @@ void Note::layout2()
             const StaffType* tab = st->staffType(tick());
             qreal mags = magS();
             bool paren = false;
-            _fretHidden = false;
             if (tieBack() && !tab->showBackTied() && !_fretString.startsWith("(")) {   // skip back-tied notes if not shown but between () if on another system
                   if (chord()->measure()->system() != tieBack()->startNote()->chord()->measure()->system() || el().size() > 0)
```

Comment:
Is the system part of this still a meaningful check?

Reply:
Yes actually, although it may be a bit confusing and/or misleading. This condition works perfectly for determining whether or not to set `paren` to `true`. If both systems are NULL, then `paren` should be `false` in case "Show back-tied fret marks" is ON.

Reply:
> If both systems are NULL, then paren should be false in case "Show back-tied fret marks" is ON.

I guess that is not really relevant after all, since this is only evaluated when `tab->showBackTied()` is `false`. It should be enough to test whether the tied notes are in the same measure. If they are in different measures, or if `el().size() > 0`, then if the fret mark is shown at all, it should be in parentheses. Come to think of it, this can all be done in `Note::layout()`, since the note's measure is already known at that point. I will update this PR accordingly.

Reply:
One could argue that when `tab->showBackTied()` is `false`, then if the fret mark is shown at all, it should be in parentheses, and we do not need to check whether the two notes are in different measures, or if `el().size() > 0`. While this is technically true, these are the cases where we might end up showing the fret mark after all, and adding parentheses in all cases would unnecessarily increase the width of the hidden fret mark, and potentially increase the width of the measure as a result.

Reply:
The idea that we could add the parens always since they are always shown (if the fret is) did occur me, but I agree that increasing the measure width in the (more common) invisible case is not good.  So it's still good to at least make some attempt to tell the difference, but to err on the side of adding them if we can't be sure of a meaningful test at the point we are doing this.  So testing different measures seems like a good compromise indeed for this case, but probably not for accidentals on tied notes on standard staves.

Looking ahead to possibly more structural fixes:

I've been working on code that is possibly relevant here, it will affect how often we call getNextMeasure() for each measure we layout.  Right now sometimes we call it twice, sometimes not at all (!), the goal is it would always be called exactly once, and I think my PR #5081 accomplishes that pretty well.  The one time it is called, though, will be when we're still trying to figure out if the measure fits on the system.  So there will be no way to get reasonable system info from anything called downstream of getNextMeasure (that's already true, except in the cases where it's getting called twice, which has to do with also trying to decide if the system fits on the page or not, and that's one of the things my PR fixes).

However, computeMinWidth() *will* be called twice for the first measure of each system within the layout range (again, that's mostly true now and will be more so after my PR is merged).  computeMinWidth() should be able to know if it is the first measure of a system.  It definitely knows if there is a header, which is also useful info - we should probably be treating measure after horizontal frames as start of system for these purposes.  And we definitely want to have the proper segment with by the time computeMinWidth() finishes.

So I'm back to thinking that the best long term fix is to have computeMinWidth - or some function that gets called from there - do these sorts of things.  For the back-tied fret mark case, for the repeat-duration-symbols-at-start-of-system case, and for the accidental on standard staff case.  Maybe a new Chord::layoutStartOfSystem() function that gets called in computeMinWidth(), and it deals with the duration symbols but then iterates over notes to deal with the tie stuff.

Reply:
I would also like to see a structural solution like the one described, but given the relative complexity of its planning and implementation I am inclined to think it would be better to merge this pull request right now to make this issue fixed in the nearest releases.

Concerning the idea of a more persistent solution, thinking more widely we can even make a special mechanism for such cases: having a structure like `LayoutAssumptions` which would contain such assumptions that cannot be safely established at some stage of layout and which would be passed to layout functions to be used in their logic. Of course, this will be a much larger change to layout system, and for performance reasons it will probably need some mechanism to define which layout stages actually depend on any temporary assumptions but maybe that could be something to consider if we get to making a layout subsystem more simple and clear in general. After all, it looks exactly like what we do right now in some cases: assume something, apply this assumption and look at the consequences.